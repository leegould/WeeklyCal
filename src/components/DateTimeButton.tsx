import React from 'react';
import { View, StyleSheet } from 'react-native';
import moment, { Moment } from 'moment';
import DateTimePicker from 'react-native-modal-datetime-picker';
import { Button, Icon } from 'react-native-elements';

export interface Props {
    date: Moment,
    showTime: boolean,
    onDateChanged: Function,
};

interface State {
    date: Moment,
    isDateTimePickerVisible: boolean,
};

export default class DateTimeButton extends React.PureComponent<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            date: props.date, //.startOf('day').add(12, 'hours'),
            isDateTimePickerVisible: false,
        }
    }

    showDateTimePicker = () => this.setState({ isDateTimePickerVisible: true });

    hideDateTimePicker = () => this.setState({ isDateTimePickerVisible: false });

    handleDatePicked = (date: Date) => {
        console.log('A date has been picked: ', date);
        this.setState({ date: moment(date) });
        this.hideDateTimePicker();
        this.props.onDateChanged(moment(date));
    };

    render() {
        return (
            <View style={styles.container}>
                <Button
                    onPress={this.showDateTimePicker as any}
                    title={this.state.date.format(this.props.showTime ? 'DD-MM-YYYY' : 'DD-MM-YY HH:mm')}
                    titleStyle={styles.title}
                    icon={<Icon name='calendar' type='material-community' color='#C2272D' size={18} />}
                    type='solid'
                    buttonStyle={styles.button}
                />
                <DateTimePicker
                    isVisible={this.state.isDateTimePickerVisible}
                    onConfirm={this.handleDatePicked}
                    onCancel={this.hideDateTimePicker}
                    date={this.state.date.toDate()}
                    mode={this.props.showTime ? 'date' : 'datetime'}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        backgroundColor: 'lightgray'
    },
    button: {
        backgroundColor: 'lightgray',
        borderWidth: 0
    },
    title: {
        marginLeft: 5,
        color: 'white',
        fontSize: 14
    },
});
