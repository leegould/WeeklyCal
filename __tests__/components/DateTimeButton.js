import React from 'react';
import renderer from 'react-test-renderer';
import DateTimeButton from '../../src/components/DateTimeButton';
import moment from 'moment';

test('renders correctly with expected args', () => {

    const props = {
        date: moment(),
        showTime: false,
        onDateChanged: () => {},
    }

    const tree = renderer.create(<DateTimeButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
