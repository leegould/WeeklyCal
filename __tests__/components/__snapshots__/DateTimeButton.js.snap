// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with expected args 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightgray",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "lightgray",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#C2272D",
                    "fontSize": 18,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ïƒ­
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "marginLeft": 5,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          31-05-19 11:19
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "margin": 10,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 13,
              "marginBottom": 8,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#d5d5d5",
              "borderBottomWidth": 0.5,
              "padding": 14,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8f8f8f",
                  "fontSize": 13,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Pick a date
          </Text>
        </View>
        <View
          onStartShouldSetResponderCapture={[Function]}
        >
          <View>
            <RCTDatePicker
              date={1559297943206}
              minuteInterval={1}
              mode="datetime"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 216,
                }
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#d5d5d5",
              "borderTopWidth": 0.5,
              "height": 57,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "normal",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Confirm
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 13,
            "height": 57,
            "justifyContent": "center",
            "marginBottom": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#007ff9",
                "fontSize": 20,
                "fontWeight": "600",
                "padding": 10,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
